File: 03_recursion/c++/01_countdown.cpp
#include <iostream>

using namespace std;

int countdown(int num){
	if(num > 0)
		cout << num << endl;
	if (num <= 0)
		return num;
	else 
		return countdown(num-1);
}

int main(int argc, char *argv[]) {
	cout << countdown(5) << endl;
}

File: 03_recursion/java/01_countdown.java
import java.util.*;
import java.lang.*;
import java.io.*;

class grokkingAlgorithms
{	
	public static int countdown(int num){
		if(num > 0)
			System.out.println(num);
		if(num <= 0)
			return num;
		else
			return countdown(num-1);
	}

	public static void main (String[] args) throws java.lang.Exception
	{
		System.out.println(countdown(5));
	}
}